MidasCivil_Adapter/CRUD/Create/Loads/BarDifferentialTemperatureLoad.cs

```diff
@@ -0,0 +1,82 @@
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using System;
+using System.Linq;
+using BH.oM.Adapters.MidasCivil;
+using BH.Engine.Adapter;
+using BH.oM.Structure.Loads;
+using BH.oM.Structure.Elements;
+using System.Collections.Generic;
+using System.IO;
+
+namespace BH.Adapter.MidasCivil
+{
+    public partial class MidasCivilAdapter
+    {
+        /***************************************************/
+        /**** Private Methods                           ****/
+        /***************************************************/
+
+        private bool CreateCollection(IEnumerable<BarDifferentialTemperatureLoad> BarDifferentialTemperatureLoads)
+        {
+            string loadGroupPath = CreateSectionFile("LOAD-GROUP");
+            foreach (BarDifferentialTemperatureLoad barDifferentialTemperatureLoad in BarDifferentialTemperatureLoads)
+            {
+                if (barDifferentialTemperatureLoad.TemperatureProfile.Keys.Count > 20)
+                {
+                    Engine.Reflection.Compute.RecordWarning("There is a maximium of 20 layers in the temperature profile");
+                    return true;
+                }
+                if (barDifferentialTemperatureLoad.Objects.Elements.GroupBy(x => x.SectionProperty).First().First().SectionProperty == null)
+                {
+                    Engine.Reflection.Compute.RecordWarning("Section Property is required for inputting differential temperature load");
+                    return true;
+                }
+                List<string> midasTemperatureLoads = new List<string>();
+                string barLoadPath = CreateSectionFile(barDifferentialTemperatureLoad.Loadcase.Name + "\\BSTEMPER");
+                string midasLoadGroup = Adapters.MidasCivil.Convert.FromLoadGroup(barDifferentialTemperatureLoad);
+
+                List<Bar> assignedElements = barDifferentialTemperatureLoad.Objects.Elements;
+                List<string> assignedBars = new List<string>();
+
+                foreach (Bar bar in assignedElements)
+                {
+                    assignedBars.Add(bar.AdapterId<string>(typeof(MidasCivilId)));
+                }             
+                    foreach (string assignedBar in assignedBars)
+                {
+                    midasTemperatureLoads.AddRange(Adapters.MidasCivil.Convert.FromBarDifferentialTemperatureLoad(barDifferentialTemperatureLoad, assignedBar, m_temperatureUnit));
+                }
```

Comment:
I am a little bit confused with the logic here - you're getting the ids out and creating a separate `BarDifferentialTemperatureLoad` for each - but you are able to assign multiple elements to a single load in MidasCivil.

What you need to do is the following:
1. Group the `BarDifferentialTemperatureLoad` by `SectionProperty`
2. For each group - you can determine whether there is a valid `SectionProperty` and proceed with the convert 
3. Then you can convert the `BarDifferentialTemperatureLoad` and pass through the `Bars` it is assigned to (perhaps just the ids and SectionPropety so you're not passing through the `Bars` themselves